### PR TITLE
[NFC] Draft of style guidelines.

### PR DESCRIPTION
**Pull request by schweitzpgi**
Update comments to point to flang/llvm style document.

### CODE REVIEW

**Review by psteinfeld — CHANGES_REQUESTED** [review]

File: flang/docs/LLVMCodingStyle.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+   Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+   See https://llvm.org/LICENSE.txt for license information.
+   SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+-->
+
+# Brief Note on Flang's LLVM style.
+
+In the directories that deal extensively with MLIR and LLVM, the
+decision was made to adopt the MLIR style and coding conventions. The
```
[the remark this answers is not in the document]
Comment (schweitzpgi) [reply flang/docs/LLVMCodingStyle.md:12]: Flang has multiple coding conventions though.

[the remark this answers is not in the document]
Comment (psteinfeld) [reply flang/docs/LLVMCodingStyle.md:12]: > 
> 
> Flang has multiple coding conventions though.
To be clear, here's what I recommend:
`In the directories that deal extensively with MLIR and LLVM, Flang uses the MLIR style and coding conventions.`

**Review by psteinfeld — APPROVED** [review]
Looks great!

Ship it.

**Review by jeanPerier — APPROVED** [review]

**Comment by banach-space** [comment]
This is a divergence from LLVM's coding style and I feel that it should be discussed with the wider community first. Unless this is to remain a `fir-dev` specific change? But IIUC, all of `fir-dev` is to be moved to LLVM's `main`.

**Comment by clementval** [comment]
> This is a divergence from LLVM's coding style and I feel that it should be discussed with the wider community first. Unless this is to remain a `fir-dev` specific change? But IIUC, all of `fir-dev` is to be moved to LLVM's `main`.

The LLVM coding style is not very clear on this (at least it is how I read it) and we are not saying use `auto` everywhere all the time. Anyway this will be probably be discussed when pushing this upstreaming in the phabricator patch.

**Review by clementval — APPROVED** [review]

**Comment by banach-space** [comment]
> @banach-space The LLVM coding style is not very clear on this (at least it is how I read it)

That's why I think it is important to involve people who wrote the original guidelines (or participated in that discussion).

> Anyway this will be probably be discussed when pushing this upstreaming in the phabricator patch.

Why not create this PR on Phabricator then? This is effectively increasing the `fir-dev` delta. Admittedly, not by much, but it does touch quite a few files. Also, we do know that there are developers actively reviewing Flang patches on Phabricator and who are interested in this conversation, but don't really visit here.

IMHO, it would make a lot of sense to send this directly to LLVM `main`.

**Comment by schweitzpgi** [comment]
> > @banach-space The LLVM coding style is not very clear on this (at least it is how I read it)
> 
> That's why I think it is important to involve people who wrote the original guidelines (or participated in that discussion).
> 
> > Anyway this will be probably be discussed when pushing this upstreaming in the phabricator patch.
> 
> Why not create this PR on Phabricator then? This is effectively increasing the `fir-dev` delta. Admittedly, not by much, but it does touch quite a few files. Also, we do know that there are developers actively reviewing Flang patches on Phabricator and who are interested in this conversation, but don't really visit here.
> 
> IMHO, it would make a lot of sense to send this directly to LLVM `main`.

I agree that more discussion can be had on mailing lists and/or phabricator.

This documents what has been done. It clarifies the policy.

**Pull request closed without merging**